iib/web/models.py

```diff
@@ -1398,3 +1400,106 @@ def validate_request_params(request_params, required_params, optional_params):
             and not request_params[param]
         ):
             del request_params[param]
+
+#TODO modify
+class RequestCreateEmptyIndex(Request, RequestIndexImageMixin):
+    """An "create-empty-index" image build request."""
+
+    __tablename__ = 'request_create_empty_index'
+
+    id = db.Column(db.Integer, db.ForeignKey('request.id'), autoincrement=False, primary_key=True)
+
+    labels = db.Column(db.String, nullable=True)
+    binary_image = db.Column(db.String, nullable=True)
+    from_index = db.Column(db.String, nullable=True)
+
+    __mapper_args__ = {'polymorphic_identity': RequestTypeMapping.__members__['create-empty-image'].value}
+
+    @classmethod
+    def from_json(cls, kwargs, batch=None):
+        """
+        Handle JSON requests for the create-empty-index API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        # Check if no bundles and `from_index is specified
+        # if no bundles and no from index then an empty index will be created which is a no-op
+        if request_kwargs.get('from_index'):
+            raise ValidationError('"from_index" must be specified')
+
+        # I am not sure if I should check it here and then
+        # still use it as an additional parameter in _from_json below
+        if not isinstance(request_kwargs.get('binary_image'), str):
+            raise ValidationError('"binary_image" must be a string')
+
+        labels = request_kwargs.get('labels', [])
+        if not isinstance(labels, list) or any(
+                not item or not isinstance(item, str) for item in labels
+        ):
+            raise ValidationError('"labels" should be either an empty array or an array of non-empty strings')
+
+
+        # Remove attributes that are not stored in the database
+        request_kwargs.pop('cnr_token', None)
```

Comment:
What is it used for?

Reply:
We do not want to store sensitive information (token in this case)